botcore/perms.py

```diff
@@ -1,83 +1,62 @@
-from discord import DMChannel
-from discord.ext.commands import CheckFailure
-import inspect
-import functools
+from functools import wraps
+from inspect import iscoroutinefunction
 
 from botcore.config import config
 
-class NotVerified(CheckFailure):
-    pass
-
-class AlreadyVerified(CheckFailure):
-    pass
-
-class NotAdminUser(CheckFailure):
-    pass
-
-class NotGuildMember(CheckFailure):
-    pass
-
-class NotAllowedChannel(CheckFailure):
-    pass
-
-class NotAdminChannel(CheckFailure):
-    pass
-
-class NotDMChannel(CheckFailure):
-    pass
-
-def check(predicate):
+def check(predicate, error):
```

Comment:
Doesn't the discord python library already provide a [check decorator](https://discordpy.readthedocs.io/en/latest/ext/commands/api.html#ext-commands-api-checks)?